include/aspect/parameters.h

```diff
@@ -308,6 +308,9 @@ namespace aspect
 
     double                         nonlinear_tolerance;
     double                         nonlinear_switch_tolerance;
+    std::string                    use_Newton_stabilisation_preconditioner;
+    std::string                    use_Newton_stabilisation_A_block;
+    bool                           use_Newton_failsafe;
```

Comment:
If we decide this is a good way, we might even omit this variable and always fall back to the failsafe.